hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/commit/BaseCommitActionExecutor.java

```diff
@@ -240,12 +241,16 @@ protected void commit(HoodieWriteMetadata<O> result, List<HoodieWriteStat> write
             updateColumnsToIndexForColumnStats(metaClient, columnsToIndex);
             return null;
           });
-    } catch (HoodieIOException e) {
+    } catch (HoodieIOException | IOException e) {
```

Comment:
What is the reason of catching `IOException` here?

Reply:
appendMetadataForMissingFiles throws IOException 
  -> CommitMetadataUtils.reconcileMetadataForMissingFiles also throws 
       -> markers.getAppendedLogPaths(context, config.getFinalizeWriteParallelism()) throws
even create marker api throws IOException.